.github/workflows/wasm-benhmarks.yml

```diff
@@ -0,0 +1,99 @@
+name: Wasm perf
+on:
+  pull_request:
+
+jobs:
+  benchmarks:
+    runs-on: ubuntu-latest
+    env: # Set environment variables for the whole job
+      PROFILE: release
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v4
+
+      - name: "Setup Node.js"
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node_version }}
+
+      - name: "Setup pnpm"
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+
+      - name: "Login to Docker Hub"
+        uses: docker/login-action@v3
+        continue-on-error: true
+        env:
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+        if: "${{ env.DOCKERHUB_USERNAME != '' && env.DOCKERHUB_TOKEN != '' }}"
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - uses: cachix/install-nix-action@v24
+
+      - name: Setup benchmark
+        run: make setup-pg-bench
+
+      - name: Run benchmarks
+        id: bench
+        run: make run-bench | tee out.txt
+
+      - name: Read benchmark results
+        id: read_results
+        run: |
+          {
+            echo 'bench_output<<EOF'
+            cat out.txt
+            echo EOF
+          } >> "$GITHUB_OUTPUT"
+
+      - name: Summarize results
+        id: summarize_results
+        run: |
+          input="${{ steps.read_results.outputs.bench_output }}"
```

Comment:
A benchmark is considered to have regressed if it has a 5% worse performance, to account for noise.